src/bootstrap/config.rs

```diff
@@ -54,7 +54,7 @@ pub struct Config {
     pub extended: bool,
     pub sanitizers: bool,
     pub profiler: bool,
-    pub ignore_git: bool,
+    pub ignore_git: Option<bool>,
```

Comment:
I don't think we should make this an optional -- I've been trying to make `Config` as much a "source of truth" as possible without needing interpretation outside this file. We can save rust.ignore_git to a variable when loading and process the logic here.